Spec/Tests/UtilsTests.swift

```diff
@@ -0,0 +1,42 @@
+import Ably
+import Nimble
+import XCTest
+
+class UtilsTests: XCTestCase {
+    
+    struct Constants {
+        static let backoffValues = [ 1.0, 4.0/3.0, 5.0/3.0, 2.0, 2.0 ]
+        static let delayValues = [ 13.5, 18.0, 22.5, 27.0, 27.0 ] // calculated for randomFraction = 0.5 (jitterCoeff = 0.9)
+    }
+    
+    // MARK: Backoff and jitter
+    
+    func testRetryDelayCalculationForRetryNumberWithConstantRandomSeeding() {
+        let initialRetryTimeout = 15.0
+        
+        ARTUtils.seedRandom(withNumber: 5)
+        let delays1 = (1...5).map { n in initialRetryTimeout * ARTUtils.backoffCoefficient(forRetryNumber: n) * ARTUtils.generateJitterCoefficient() }
+        
+        ARTUtils.seedRandom(withNumber: 5)
+        let delays2 = (1...5).map { n in ARTUtils.retryDelay(fromInitialRetryTimeout: initialRetryTimeout, forRetryNumber: n) }
+        
+        expect(delays1).to(equal(delays2))
+    }
+    
+    func testRetryDelayCalculationForRetryNumberWithProperRandomSeeding() {
+        ARTUtils.seedRandomWithSystemTime()
+        
+        let initialRetryTimeout = 15.0
+        let delays = (1...5).map { n in ARTUtils.retryDelay(fromInitialRetryTimeout: initialRetryTimeout, forRetryNumber: n) }
+        
+        for i in 0..<delays.count {
+            let delay = delays[i]
+            let targetValue = Constants.delayValues[i]
+            let backoff = Constants.backoffValues[i]
+            let maxValue = initialRetryTimeout * backoff
+            let minValue = initialRetryTimeout * backoff * (1.0 - ARTUtils.jitterDelta())
+            let tolerance = (maxValue - minValue) / 2.0
+            expect(delay).to(beCloseTo(targetValue, within: tolerance))
```

Comment:
I'm a bit confused about the purpose of the `Constants.delayValues` array here – instead of this assertion could we not assert that `delay` is between `minValue` and `maxValue`?

Reply:
Thanks, is there a one-liner for `expect` for this, just curious.

Reply:
According to [the docs](https://quick.github.io/Nimble/#collection-membership) you can do `expect(5).to(beWithin(1...10))`, but that method doesn't seem to exist for me (we're on quite an old version of Nimble), so I guess just `expect((minValue...maxValue).contains(delay)).to(beTrue())`?

Reply:
Neat solution
https://github.com/ably/ably-cocoa/commit/7b35df981bdb2e1cd61012a161f1072aa48d3ae3